pinot-query-planner/src/main/java/org/apache/calcite/rel/rules/PinotJoinExchangeNodeInsertRule.java

```diff
@@ -63,7 +65,13 @@ public void onMatch(RelOptRuleCall call) {
     RelNode rightExchange;
     JoinInfo joinInfo = join.analyzeCondition();
 
-    if (joinInfo.leftKeys.isEmpty()) {
+    boolean isColocatedJoin = PinotHintStrategyTable.containsHintOption(join.getHints(),
```

Comment:
Add extensive test for the hint config + singleton exchange (also add for dynamic broadcast join)

Reply:
done